### PR TITLE
Add the metadata to the backbones

### PR DESCRIPTION
**Pull request by haifeng-jin**
- add the resnet metadata
- add csp dark net metadata

### CODE REVIEW

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py [hunk 1]
```diff
@@ -40,6 +43,9 @@
                 "[1, 3, 3, 1] depths where the batch normalization "
                 "and SiLU activation are applied after the convolution layers."
             ),
+            "params": 4223488,
```
Comment (LukeWood) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:46]: perhaps we should have task metadata to embed in the table as well!

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py [hunk 2]
```diff
@@ -22,6 +22,9 @@
                 "[1, 3, 3, 1] depths where the batch normalization "
                 "and SiLU activation are applied after the convolution layers."
             ),
+            "params": 2380416,
+            "official_name": "CSP_DARK_NET",
```
Comment (jbischof) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:26]: The official name is just what we want to display for the model. Maybe camel case would look nicer: `"CSPDarkNet"`

File: keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets.py [hunk 3]
```diff
@@ -59,6 +65,9 @@
                 "and ReLU activation are applied after the convolution layers "
                 "(v1 style)."
             ),
+            "params": 23561152,
+            "official_name": "RESNET_V1",
```
Comment (jbischof) [inline keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets.py:69]: `"ResNetV1"`?

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py [hunk 4]
```diff
@@ -314,7 +314,7 @@ def __new__(
     def presets(cls):
         """Dictionary of preset names and configurations."""
         return {
-            "cspdarknetl_imagenet": copy.deepcopy(
+            "csp_dark_net_l_imagenet": copy.deepcopy(
```
[the remark this answers is not in the document]
Comment (haifeng-jin) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py:317]: Need @jbischof input on this.

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py (hunk 1 above)
Comment (haifeng-jin) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:46]: I assume luke means add a list of `Task` subclasses, which the backbone class is compatible with, as a field of the metadata. Not all backbones compatible with all tasks?

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py (hunk 2 above)
Comment (haifeng-jin) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py]: Done.

File: keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets.py (hunk 3 above)
Comment (haifeng-jin) [reply keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_presets.py]: Done.

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbischof) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py:317]: Yes it should be "resnet", "retinanet", "darknet". Let's unify

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py [hunk 5]
```diff
@@ -40,6 +43,9 @@
                 "[1, 3, 3, 1] depths where the batch normalization "
                 "and SiLU activation are applied after the convolution layers."
             ),
+            "params": 4223488,
+            "official_name": "CSPDarkNet",
+            "path": "csp_dark_net",
```
Comment (jbischof) [inline keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py:48]: `"csp_darknet"`?

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (haifeng-jin) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py]: Done.

File: keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py (hunk 5 above)
Comment (haifeng-jin) [reply keras_cv/models/backbones/csp_darknet/csp_darknet_backbone_presets.py]: Done.

**Pull request merged**